src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -1977,6 +1977,13 @@ int sockinfo_tcp::connect(const sockaddr *__to, socklen_t __tolen)
 		return -1;
 	}
 	m_p_connected_dst_entry->prepare_to_send(false, true);
+	if (!m_p_connected_dst_entry->is_valid()) {
+		setPassthrough();
```

Comment:
You should not un-offload this socket just because a destination is unresolved.
 (don't call setPassthrough())